src/Core/SonarLintNotifications/NotificationAction.cs

```diff
@@ -0,0 +1,40 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+using System;
+
+namespace SonarLint.VisualStudio.Core.SonarLintNotifications
+{
+    public interface INotificationAction
+    {
+        string Message { get; }
+        Action Action { get; }
```

Comment:
The gold bar can show buttons and hyperlinks. This API doesn't give a way for the caller to specify which.

Do we think it needs to? Stylistically, MS-VS gold bars seem to use a mix of buttons and links, but my impression is that the newer gold bars just use links.

Reply:
It might make sense to show a button for main actions. For instance for nodejs we can show something like "install node" and making that button makes sense while "more info" and "don't show this again" would be better as links.